libnmstate/nm/bridge_port_vlan.py

```diff
@@ -0,0 +1,251 @@
+        API.
+        :param trunk_tags: list of schema.LinuxBridge.Port.Vlan.TrunkTags
+               objects.
+        :param tag: the access tag for access ports, the native vlan ID for
+               trunk ports
+        :param is_native_vlan: boolean attribute indicating if the trunk port
+               has a native vlan.
+        """
+        self._trunk_ports = trunk_tags
+        self._tag = tag
+        self._is_native_vlan = is_native_vlan
+        self._port_type = (
+            LB.Port.Vlan.Mode.TRUNK if trunk_tags else LB.Port.Vlan.Mode.ACCESS
+        )
+
+    @property
+    def trunk_ports(self):
+        return self._trunk_ports
+
+    @property
+    def tag(self):
+        return self._tag
+
+    @property
+    def enable_native_vlan(self):
+        return self._is_native_vlan
+
+    @property
+    def port_type(self):
+        return self._port_type
+
+    def to_nm(self):
+        """
+        Generate a list of NM.BridgeVlan objects from the encapsulated
+        PortVlanFilter data
+        """
+
+        port_vlan_config = []
+        if self.port_type == LB.Port.Vlan.Mode.TRUNK:
+            for trunk_port in self.trunk_ports:
+                port_vlan_config.append(
+                    PortVlanFilter._generate_vlan_trunk_port_config(trunk_port)
+                )
+            if self.enable_native_vlan and self.tag:
+                port_vlan_config.append(
+                    PortVlanFilter._generate_vlan_access_port_config(self.tag)
+                )
+        elif self.port_type == LB.Port.Vlan.Mode.ACCESS and self.tag:
+            port_vlan_config.append(
+                PortVlanFilter._generate_vlan_access_port_config(self.tag)
+            )
+
+        return port_vlan_config
+
+    def to_dict(self):
+        """
+        Get the port vlan filtering configuration in dict format - e.g. in yaml
+        format:
+        - name: eth1
+          vlan:
+            type: trunk
+            trunk-tags:
+              - id: 101
+              - id-range:
+                  min: 200
+                  max: 4095
+            tag: 100
+            enable-native: true
+        """
+
+        port_vlan_state = {
+            LB.Port.Vlan.MODE: self.port_type,
+            LB.Port.Vlan.TRUNK_TAGS: self.trunk_ports,
+        }
+        if self.tag:
+            port_vlan_state[LB.Port.Vlan.TAG] = self.tag
+        if self.port_type == LB.Port.Vlan.Mode.TRUNK:
+            port_vlan_state[
+                LB.Port.Vlan.ENABLE_NATIVE
+            ] = self.enable_native_vlan
+        return port_vlan_state
+
+    def import_from_bridge_settings(self, port_vlans):
+        """
+        Instantiates a PortVlanFilter object from a list of NM.BridgeVlan
+        objects.
+        """
+
+        bridge_vlan_config = [
+            PortVlanFilter._get_vlan_info(port_vlan_filter_entry)
+            for port_vlan_filter_entry in port_vlans
+        ]
+
+        trunk_tags, tag, enable_native = PortVlanFilter._get_vlan_config(
```

Comment:
"get_vlan_info" and "get_vlan_config" seem very similar to me and I do not really understand what each does.
One seems to read the nm bridge vlan setting and place it in a dict, then the other uses it to convert it to another data struct (actually 3 new ones).

Bottom line, I do not understand the separation here and the naming is not helping.
If there is no good name for these, perhaps the abstraction needs fixing.
This last function also returns 3 items, all part of the object state itself. So it is unclear why it did not updated directly the state.